crates/lox_io/src/spice.rs

```diff
@@ -0,0 +1,369 @@
+use nom::branch::alt;
+use nom::bytes::complete::{tag, take_until, take_while1};
+use nom::character::complete::{alpha1, line_ending, multispace0, one_of};
+use nom::combinator::{map, map_res, recognize, rest};
+use nom::multi::{fold_many0, fold_many1, many0, many1};
+use nom::number::complete::{double, float};
+use nom::sequence::{delimited, preceded, separated_pair, terminated, tuple};
+use nom::IResult;
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq)]
+enum Value {
+    Double(f64),
+    String(String),
+    DoubleArray(Vec<f64>),
+    StringArray(Vec<String>),
+}
+
+pub struct Kernel {
+    type_id: String,
+    items: HashMap<String, Value>,
+}
+
+type Entries = Vec<(String, Value)>;
+
+impl Kernel {
+    pub fn parse(input: &str) -> Result<Self, &'static str> {
```

Comment:
Prefer an enum instead of the &str error type

Reply:
I would like to do something with the error returned by `nom` to provide a meaningful error message to users but would also like to defer that.